base/set.jl

```diff
@@ -476,7 +476,40 @@ allunique(::Union{AbstractSet,AbstractDict}) = true
 
 allunique(r::AbstractRange) = !iszero(step(r)) || length(r) <= 1
 
-allunique(A::StridedArray) = length(A) < 32 ? _indexed_allunique(A) : _hashed_allunique(A)
+function allunique(A::StridedArray)
+    if length(A) < 32
+        _indexed_allunique(A)
+    elseif OrderStyle(eltype(A)) === Ordered()
```

Comment:
Perhaps the branch to `_indexed_allunique` should be later, as this is faster than `_indexed_allunique` on short sorted arrays, not only faster than `_hashed_allunique`. Some quick times, although more care would be needed e.g. about short-but-not-sorted cases:
```julia
julia> let xs = collect(reverse(1:100))  # case this PR aims to improve
         @btime allunique($xs) 
         @btime pr_allunique($xs)
       end;
  min 860.368 ns, mean 1.288 μs (10 allocations, 3.62 KiB)
  min 34.875 ns, mean 35.051 ns (0 allocations)

julia> let xs = collect(reverse(1:10))
         @btime allunique($xs)     # uses _indexed_allunique
         @btime pr_allunique($xs)  # PR modified to never use _indexed_allunique
       end;
  min 25.184 ns, mean 25.409 ns (0 allocations)
  min 6.791 ns, mean 6.893 ns (0 allocations)

julia> let xs = randn(100)  # this always goes to _hashed_allunique
         @btime allunique($xs) 
         @btime pr_allunique($xs)
       end;
  min 1.333 μs, mean 1.736 μs (10 allocations, 3.62 KiB)
  min 1.342 μs, mean 1.756 μs (10 allocations, 3.62 KiB)
```

Reply:
It does take longer for small, unsorted arrays. I'd call it a wash, especially considering the uncertainty around the distribution of the inputs